src/density.js

```diff
@@ -46,6 +48,21 @@ export default function() {
         values[x0 + 1 + y0 * n] += xt * (1 - yt) * wi;
         values[x0 + 1 + (y0 + 1) * n] += xt * yt * wi;
         values[x0 + (y0 + 1) * n] += (1 - xt) * yt * wi;
+
+        if(aggregation === 'MEAN') {
+          valueCounts[x0 + y0 * n] += 1;
+          valueCounts[x0 + 1 + y0 * n] += 1;
+          valueCounts[x0 + 1 + (y0 + 1) * n] += 1;
+          valueCounts[x0 + (y0 + 1) * n] += 1;
+        }
+      }
+    }
+
+    if(aggregation === 'MEAN') {
+      for (i = 0; i < n * m; i++) {
+        if (valueCounts[i]) {
+          values[i] /= valueCounts[i];
```

Comment:
what is the mean in locations where there is no data around? Should it be NaN, or 0?

Reply:
I think zero makes sense when the values are non-negative. It appears that is the assumption in DeckGL https://github.com/visgl/deck.gl/blob/f4a9afc07b925c5c9bb6b53d96d15541a2f0fb4f/modules/aggregation-layers/src/utils/aggregation-operation-utils.ts#L46 I suppose ideally there should be a way to override that with a different value, like `.emptyValue(-Infinity)`